source/slangc/main.cpp

```diff
@@ -26,14 +27,10 @@ static void diagnosticCallback(
 #define MAIN main
 #endif
 
-// Used to identify that compilation was the failure - with a unique 'internal' code
-#define SLANG_E_INTERNAL_COMPILE_FAILED SLANG_MAKE_ERROR(SLANG_FACILITY_INTERNAL, 0x7fab)
-
-static SlangResult innerMain(int argc, char** argv)
+SLANG_SHARED_LIBRARY_TOOL_API SlangResult innerMain(AppContext* appContext, SlangSession* session, int argc, const char*const* argv)
 {
-    // Parse any command-line options
+    AppContext::setSingleton(appContext);
```

Comment:
I get why these are handled as a singleton/global, but it also seems a bit messy when we are also invoking a bunch of different `innerMain`s as part of testing.

One random thought: should these I/O channels actually be handled as thread-locals rather than true globals? That would get messier whenever we start to multithread anything (including the compiler) because we'd have to deal with inheriting that state across threads, so I'm not actually advocating for it any time soon.